circuit_maintenance_parser/parsers/aws.py

```diff
@@ -23,10 +23,26 @@ def parse_subject(self, subject):
 
         Example: AWS Direct Connect Planned Maintenance Notification [AWS Account: 00000001]
         """
-        data = {}
-        search = re.search(r"\[AWS Account ?I?D?: ([0-9]+)\]", subject)
-        if search:
-            data["account"] = search.group(1)
+        data = {"account": ""}
+        # Common Subject strings for matching:
+        subject_map = {
+            r"\[AWS Account ?I?D?: ([0-9]+)\]": "account",
+        }
+
+        regex_keys = re.compile("|".join(subject_map), re.IGNORECASE)
+
+        # in case of a multi-line subject
+        # match the subject map
+        for line in subject.splitlines():
+            line_matched = re.search(regex_keys, line)
+            if not line_matched:
+                continue
+            for group_match in line_matched.groups():
+                if not group_match:
+                    continue
+                for key, value in subject_map.items():
+                    if re.search(key, line, re.IGNORECASE):
+                        data[value] = group_match
```

Comment:
I know it works and could cover more future REGEX matching besides "account", but I guess it was not necessary to introduce a map for this unique key, no? (no need to change)

Reply:
Yea, when I was extending the text parser, i realized I could do the same thing on subject, and thought that it might make sense just to do the same on both parsers, that way, in case this case ever happened again, it would just be a quick regex pattern match. ... But then after doing it, I realized that there was only the match.... but I by that time I was committed, plus it felt like sense to keep the same pattern in the provider.